rln/src/public.rs

```diff
@@ -830,6 +830,77 @@ impl RLN<'_> {
+    ///
+    /// let mut input_proof_data_1 = Cursor::new(proof_data_1);
+    /// let mut input_proof_data_2 = Cursor::new(proof_data_2);
+    /// let mut output_buffer = Cursor::new(Vec::<u8>::new());
+    /// rln.recover_id_secret(
+    ///     &mut input_proof_data_1,
+    ///     &mut input_proof_data_2,
+    ///     &mut output_buffer,
+    /// )
+    /// .unwrap();
+    ///
+    /// let serialized_id_secret = output_buffer.into_inner();
+    ///
+    /// // We ensure that a non-empty value is written to output_buffer
+    /// assert!(!serialized_id_secret.is_empty());
+    ///
+    /// // We check if the recovered id secret corresponds to the original one
+    /// let (recovered_id_secret, _) = bytes_le_to_fr(&serialized_id_secret);
+    /// assert_eq!(recovered_id_secret, identity_secret);
+    /// ```
+    pub fn recover_id_secret<R: Read, W: Write>(
+        &self,
+        mut input_proof_data_1: R,
+        mut input_proof_data_2: R,
+        mut output_data: W,
+    ) -> io::Result<()> {
+        // We deserialize the two proofs and we get the corresponding RLNProofValues objects
+        let mut serialized: Vec<u8> = Vec::new();
+        input_proof_data_1.read_to_end(&mut serialized)?;
+        // We skip deserialization of the zk-proof at the beginning
+        let (proof_values_1, _) = deserialize_proof_values(&serialized[128..].to_vec());
+
+        let mut serialized: Vec<u8> = Vec::new();
+        input_proof_data_2.read_to_end(&mut serialized)?;
+        // We skip deserialization of the zk-proof at the beginning
+        let (proof_values_2, _) = deserialize_proof_values(&serialized[128..].to_vec());
+
+        // We continue only if the proof values are for the same epoch
+        // The idea is that proof values that go as input to this function are verified first (with zk-proof verify), hence ensuring validity of epoch and other fields.
+        // Only in case all fields are valid, an external_nullifier for the message will be stored (otherwise signal/proof will be simply discarded)
+        // If the nullifier matches one already seen, we can recovery of identity secret.
+        if proof_values_1.epoch == proof_values_2.epoch {
+            // We extract the two shares
+            let share1 = (proof_values_1.x, proof_values_1.y);
+            let share2 = (proof_values_2.x, proof_values_2.y);
+
+            // We recover the secret
+            let recovered_id_secret = compute_id_secret(share1, share2, proof_values_1.epoch);
+
+            // If an id secret is recovered, we write it to output_data, otherwise nothing will be written.
+            if recovered_id_secret.is_ok() {
+                let id_secret = recovered_id_secret.unwrap();
+                output_data.write_all(&fr_to_bytes_le(&id_secret))?;
+            }
+        }
+
+        return Ok(());
```

Comment:
```suggestion
        Ok(())
```
Based on other functions in this module :)